contracts/NonFungibleToken-v2.cdc

```diff
@@ -0,0 +1,158 @@
+    }
+
+    /// Interface to mediate withdraws from the Collection
+    ///
+    pub resource interface Transferable {
+        /// withdraw removes an NFT from the collection and moves it to the caller
+        pub fun transfer(id: UInt64, receiver: Capability<&AnyResource{Receiver}>): @AnyResource{NFT}
+    }
+
+    /// Interface to mediate deposits to the Collection
+    ///
+    pub resource interface Receiver {
+
+        /// deposit takes an NFT as an argument and adds it to the Collection
+        ///
+        pub fun deposit(token: @AnyResource{NFT})
+
+        /// getAcceptedTypes returns a list of NFT types that this receiver accepts
+        pub fun getAcceptedTypes(): [Type]
+    }
+
+    /// Interface that an account would commonly 
+    /// publish for their collection
+    pub resource interface CollectionPublic { //: MetadataViews.ResolverCollection {
+        pub fun deposit(token: @AnyResource{NFT})
+        pub fun borrowViewResolver(id: UInt64): &{MetadataViews.Resolver}
+        pub fun getIDs(): [UInt64]
+        pub fun borrowNFT(id: UInt64): &AnyResource{NFT}
+    }
+
+    /// Requirement for the concrete resource type
+    /// to be declared in the implementing contract
+    ///
+    pub resource interface Collection { //: Provider, Receiver, Transferable, CollectionPublic, MetadataViews.ResolverCollection {
+
+        /// Paths for the collection
+        pub let storagePath: StoragePath
+        pub let publicPath: PublicPath
+        pub let privateProviderPath: PrivatePath
+
+        /// Dictionary to hold the NFTs in the Collection
+        /// Normally we would require that the collection specify
+        /// a specific dictionary to store the NFTs, but this isn't necessary any more
+        /// as long as all the other functions are there
+
+        /// Returns the NFT types that this collection can store
+        pub fun getAcceptedTypes(): [Type]
+
+        /// withdraw removes an NFT from the collection and moves it to the caller
+        pub fun withdraw(withdrawID: UInt64): @AnyResource{NFT}
+
+        /// deposit takes a NFT and adds it to the collections dictionary
+        /// and adds the ID to the id array
+        pub fun deposit(token: @AnyResource{NFT})
+
+        /// Function for a direct transfer instead of having to do a deposit and withdrawal
+        ///
+        pub fun transfer(id: UInt64, recipient: Capability<&{NonFungibleToken.Receiver}>): Bool
+
+        /// getIDs returns an array of the IDs that are in the collection
+        pub fun getIDs(): [UInt64]
+
+        /// Returns a borrowed reference to an NFT in the collection
+        /// so that the caller can read data and call methods from it
+        pub fun borrowNFT(id: UInt64): &AnyResource{NonFungibleToken.NFT}?
+
+        /// From the MetadataViews Contract
+        /// borrows a reference to get metadata views for the NFTs that the contract contains
+        pub fun borrowViewResolver(id: UInt64): &{MetadataViews.Resolver}
+
```

Comment:
do we need this?

Reply:
Yeah, it is part of the `MetadataViews.ResolverCollection` interface